### PR TITLE
Apple m5120 keyboard

### PR DESCRIPTION
**Pull request by Dbroqua**
This is just a simple use of Teensy on an old Apple Keyboard.

## Description

I used a Teensy controller to bring up a new life of an old Apple laptop keyboard.

Not sure it's really useful to push this work on QMK but I don't know where save and share this work.

The goal is to complete this work with more documentation if some people interested.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/apple_m5120/iso/config.h [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#pragma once
```
Comment (drashna) [inline keyboards/apple_m5120/iso/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/apple_m5120/iso/config.h [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+#include "config_common.h"
+
+#define VENDOR_ID 0xFEED
+#define PRODUCT_ID 0x991D
+#define DEVICE_VER 0x0001
+#define MANUFACTURER Apple
+#define PRODUCT M5120
+#define DESCRIPTION Apple laptop keyboard controlled by a Teensy
```
Comment (drashna) [inline keyboards/apple_m5120/iso/config.h:10]:
```suggestion
```

File: keyboards/apple_m5120/iso/iso.c [hunk 3]
```diff
@@ -0,0 +1 @@
+#include "iso.h"
```
Comment (drashna) [inline keyboards/apple_m5120/iso/iso.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/apple_m5120/iso/iso.h [hunk 4]
```diff
@@ -0,0 +1,23 @@
+#pragma once
```
Comment (drashna) [inline keyboards/apple_m5120/iso/iso.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/apple_m5120/iso/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,24 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/apple_m5120/iso/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/apple_m5120/iso/rules.mk [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+NKRO_ENABLE = no
+BACKLIGHT_ENABLE = no
+RGBLIGHT_ENABLE = no
+MIDI_ENABLE = no
+BLUETOOTH_ENABLE = no
+AUDIO_ENABLE = no
+FAUXCLICKY_ENABLE = no
+HD44780_ENABLE = no
+SPLIT_KEYBOARD = no
+IOS_DEVICE_ENABLE = no
```
Comment (drashna) [inline keyboards/apple_m5120/iso/rules.mk:20]:
```suggestion
# MCU name
MCU = atmega32u4

# Bootloader selection
BOOTLOADER = halfkay

# Build Options
#   change to "no" to disable the options, or define them in the Makefile in
#   the appropriate keymap folder that will get included automatically
#
BOOTMAGIC_ENABLE = yes
MOUSEKEY_ENABLE = no
EXTRAKEY_ENABLE = no
CONSOLE_ENABLE = no
COMMAND_ENABLE = no
SLEEP_LED_ENABLE = no
NKRO_ENABLE = no
BACKLIGHT_ENABLE = no
RGBLIGHT_ENABLE = no
MIDI_ENABLE = no
BLUETOOTH_ENABLE = no
AUDIO_ENABLE = no
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/apple_m5120/iso/config.h [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+  */
+
+#pragma once
+
+#include "config_common.h"
+
+#define VENDOR_ID 0xFEED
+#define PRODUCT_ID 0x991D
+#define DEVICE_VER 0x0001
+#define MANUFACTURER Apple
```
Comment (fauxpark) [inline keyboards/apple_m5120/iso/config.h:24]: May want to change this. You are not Apple.

File: keyboards/apple_m5120/iso/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+   * |--------------------------------------------------------------------------------------+
+   * | Ctrl| GUI    | LALT    |               Space            | RALT     | LEFT|RIGHT| DOWN|
+   * `--------------------------------------------------------------------------------------'
+   */
+  [0] = LAYOUT( /* Basic QWERTY */
+    KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, \
+    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_ENT, \
+    KC_LSFT, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_GRV, \
+    KC_LSFT, KC_BSLS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_UP,      \
+    KC_LCTL, KC_LGUI, KC_LALT, KC_SPC,  KC_RALT, KC_LEFT, KC_RIGHT, KC_DOWN \
```
Comment (fauxpark) [inline keyboards/apple_m5120/iso/keymaps/default/keymap.c:38]:
```suggestion
    KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,
    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_ENT,
    KC_LSFT, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_GRV,
    KC_LSFT, KC_BSLS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_UP,
    KC_LCTL, KC_LGUI, KC_LALT, KC_SPC,  KC_RALT, KC_LEFT, KC_RIGHT, KC_DOWN
```

File: keyboards/apple_m5120/iso/rules.mk [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+EXTRAKEY_ENABLE = no
+CONSOLE_ENABLE = no
+COMMAND_ENABLE = no
+SLEEP_LED_ENABLE = no
+NKRO_ENABLE = no
+BACKLIGHT_ENABLE = no
+RGBLIGHT_ENABLE = no
+MIDI_ENABLE = no
+BLUETOOTH_ENABLE = no
+AUDIO_ENABLE = no
```
Comment (fauxpark) [inline keyboards/apple_m5120/iso/rules.mk:20]:
```suggestion
# MCU name
MCU = atmega32u4

# Bootloader selection
BOOTLOADER = halfkay

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = full     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = no        # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/apple_m5120/iso/config.h (hunk 7 above)
Comment (Dbroqua) [reply keyboards/apple_m5120/iso/config.h:24]: I'm not Apple but the original Keyboard is from Apple and I use his PCB + Teensy. I'm not the creator of the Keyboard.

I need to set what ?

- my name? I'm not the PCB designer
- Apple? But the controler is a Teensy and the wiring is from my mind
- Teensy? But the PCB is from Apple/Alps
- My name + Apple/Alps + Teensy ?

I have no idea of what is the best.

File: keyboards/apple_m5120/iso/keymaps/default/keymap.c (hunk 8 above)
Comment (Dbroqua) [reply keyboards/apple_m5120/iso/keymaps/default/keymap.c:38]: Why? I have copy/paste this from all other keyboards/keymaps.

It's a new way to write keymaps?

File: keyboards/apple_m5120/iso/config.h (hunk 7 above)
Comment (fauxpark) [reply keyboards/apple_m5120/iso/config.h:24]: I guess this is technically a handwired board then, and so it should go in `handwired/` and have your name.

File: keyboards/apple_m5120/iso/keymaps/default/keymap.c (hunk 8 above)
Comment (fauxpark) [reply keyboards/apple_m5120/iso/keymaps/default/keymap.c:38]: The backslashes are not necessary.

File: keyboards/apple_m5120/iso/config.h (hunk 7 above)
Comment (Dbroqua) [reply keyboards/apple_m5120/iso/config.h:24]: It's a mix, original PCB with Teensy on his J1 connector (halfway between handwired and usb converter).

https://pics.darkou.fr/H3vPfmUg.png

**Comment by noroadsleft** [comment]
Sooo... there's no changes in this PR anymore. :confused:

**Comment by drashna** [comment]
This commit killed it: 685365ad7c84641310dc825de3d2bf34c247b635

**Comment by Dbroqua** [comment]
Oops…

Well that's okay. I think I would be the only one using this keyboard anyway.

You can close this PR. 

Regards

**Comment by drashna** [comment]
Well, you can revert that change, and it would fix things, but I'll go ahead and close the PR.  Feel free to re-open, if you would like.

**Pull request closed without merging**